Src/AutoFixture.NUnit3/AutoDataAttribute.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Builders;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// This attribute uses AutoFixture to generate values for unit test parameters. 
+    /// This implementation is based on TestCaseAttribute of NUnit3
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AutoDataAttribute : Attribute, ITestBuilder
+    {
+        private readonly IFixture _fixture;
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/>
+        /// </summary>
+        public AutoDataAttribute()
+            : this(new Fixture())
+        {
+        }
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/> with an <see cref="IFixture"/> 
+        /// </summary>
+        /// <param name="fixture"></param>
+        protected AutoDataAttribute(IFixture fixture)
```

Comment:
Oh, I'm sorry. — What I mean is that, for AutoFixture.Xunit2, we should make the contructor protected in `v4` because it'll be a breaking change if we do in that `master`. We did something similar already in Pull Request #462.